model/host/host_test.go

```diff
@@ -4885,3 +4885,35 @@ func TestPartitionParents(t *testing.T) {
 	assert.Len(matched, 0)
 	assert.Len(notMatched, 0)
 }
+
+func TestCountVirtualWorkstationsByDistro(t *testing.T) {
+	require.NoError(t, db.ClearCollections(Collection))
+	for i := 0; i < 100; i++ {
+		h := &Host{
+			Id:           fmt.Sprintf("%d", i),
+			Status:       evergreen.HostTerminated,
+			InstanceType: "foo",
+		}
+		if i%3 == 0 {
+			h.Status = evergreen.HostRunning
+		}
+		if i%5 == 0 {
+			h.IsVirtualWorkstation = true
+		}
+		if i%11 == 0 {
+			h.InstanceType = "bar"
+		}
+		require.NoError(t, h.Insert())
+	}
+	count, err := CountVirtualWorkstationsByInstanceType()
+	require.NoError(t, err)
+	for _, counter := range count {
```

Comment:
foo bar!
I personally found it hard to know what to expect, though.

Reply:
Yeah, I wanted it to be complex enough to do something interest, so I wrote the aggregation by hand to figure out what would happen before testing the code. I had written the aggregation in Slack already, so I was testing it against that.